tests/cookbook/runner.py

```
@@ -0,0 +1,258 @@
+            raise ContractError(f"{name}: configure exactly one of command or driver")
+        if not contract.get("expect_stdout") and not contract.get("expect_files"):
+            raise ContractError(f"{name}: contract needs at least one observable assertion")
+        if runtime == "python" and not (COOKBOOK / name / "requirements.txt").is_file():
+            raise ContractError(f"{name}: requirements.txt is missing")
+        if runtime == "typescript" and not (COOKBOOK / name / "package.json").is_file():
+            raise ContractError(f"{name}: package.json is missing")
+
+
+def _run_checked(command: list[str], cwd: Path) -> None:
+    print(f"+ {' '.join(command)}", flush=True)
+    subprocess.run(command, cwd=cwd, check=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)